solr/core/src/java/org/apache/solr/cloud/ReplicateFromLeader.java

```diff
@@ -88,31 +87,9 @@ public void startReplication(boolean switchTransactionLog) {
       log.info("Will start replication from leader with poll interval: {}", pollIntervalStr);
 
       NamedList<Object> followerConfig = new NamedList<>();
-      followerConfig.add("fetchFromLeader", Boolean.TRUE);
-
-      // don't commit on leader version zero for PULL replicas as PULL should only get its index
-      // state from leader
-      boolean skipCommitOnLeaderVersionZero = switchTransactionLog;
```

Comment:
When we are in this method, either we are a TLOG replica and `switchTransactionLog` is true, hence `skipCommitOnLeaderVersionZero` is true, or we are a PULL replica and `switchTransactionLog` is false but we force `skipCommitOnLeaderVersionZero` to true below...